glados/llama.py

```diff
@@ -34,33 +121,8 @@ def start(self, model=None, use_gpu=False):
 
         return self.is_running()
 
-    def is_running(self):
-        if self.process is not None:
-            attempts = 0
-            while True:
-                try:
-                    response = requests.get("http://localhost:8080/health")
-
-                    if response.status_code == 200:
-                        return True
-                    elif (
-                        response.status_code == 503
-                    ):  # model is still being loaded, or at full capacity
-                        pass
-                    elif response.status_code == 500:  # model failed to load
-                        self.stop()  # stop the server
-                        return False
-                    else:  # server is not running
-                        return False
-
-                except requests.exceptions.ConnectionError:
-                    attempts += 1
-                    if attempts > 10:
-                        return False
-                time.sleep(0.1)
-        return False
-
     def stop(self):
-        self.process.terminate()
-        self.process.wait()
+        if self.process:
+            self.process.terminate()
+            self.process.wait()
```

Comment:
Nice catch

Reply:
The python exception handler caught it; I just worked around it :D  There's probably a deeper race condition here, or something like that, but this makes it a bit more robust for now.